packages/common/src/api/tan-query/useUserByParams.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect } from 'react'
+
+import { useDispatch } from 'react-redux'
+
+import { ID } from '~/models/Identifiers'
+import { profilePageActions } from '~/store/pages'
+
+import { Config } from './types'
+import { useUser } from './useUser'
+import { useUserByHandle } from './useUserByHandle'
+
+const { fetchProfileSucceeded } = profilePageActions
+
+type UserParams = { userId?: ID } | { handle?: string }
+
+/**
+ * Hook that returns user data given either a user ID or handle.
+ * Uses useUser and useUserByHandle internally to maintain consistent caching behavior.
+ * @param params The user params - either {id} or {handle}
+ * @param options Optional configuration for the query
+ * @returns The user data or null if not found
+ */
+export const useUserByParams = (
+  params: UserParams | null | undefined,
+  options?: Config
+) => {
+  const dispatch = useDispatch()
+
+  const idQuery = useUser(params && 'userId' in params ? params.userId : null, {
+    ...options,
+    enabled: options?.enabled !== false && !!params && 'userId' in params
+  })
+
+  const handleQuery = useUserByHandle(
+    params && 'handle' in params ? params.handle : null,
+    {
+      ...options,
+      enabled: options?.enabled !== false && !!params && 'handle' in params
+    }
+  )
+
+  const query = 'userId' in (params ?? {}) ? idQuery : handleQuery
+  const { isSuccess, data } = query
+  const userId = data?.user_id
+
+  useEffect(() => {
```

Comment:
also wondering if this hook is specific to profile page(s) and simply calls the query hooks, does it really belong in `/tan-query`? I'm on the fence about this. I could see leaving it here or pushing it closer to the component.

I do think we should have a clear separation between the core library and fancy usages built on top of it, but this one walks the line imo.

Reply:
i can see that, my hope is that this effect eventually calls other tan-query stuff instead of saga, so the idea was to just prep for that now. lmk what you think.

Reply:
yeah more thinking like there are multiple layers to the fetching. There's fetching that is basic data utility and used lots of places, and then more complex or specific fetching logic built on top of those.

Here, I feel like getTrackById and getTrackByPermalink are both clearly base level. A custom hook that composes useTrack useUser and useCollection calls to get everything you need for a specific component all wrapped into a single call feels like it maybe should live adjacent to the component as opposed to the common library of hooks. getTrackByPageParams is borderline imo and is fine to include as a common utility, just thinking that there may be cases we want to push down the tree eventually.

Reply:
oh i see, but how do we share across web and mobile? maybe have it outside the tan-query directory but still in common. i was also going to bring up soon that potentially we improve the layout of tan-query directory, potentially following directory structure similar to sdk? ie track, user, collection have it's own directory?

Reply:
yeah we probably just need better organization